lib/redlock/client.rb

```diff
@@ -210,10 +211,16 @@ def get_remaining_ttl(resource)
 
       private
 
+      def generate_script_shas
+        @unlock_script_sha = Digest::SHA1.hexdigest(UNLOCK_SCRIPT)
+        @lock_script_sha   = Digest::SHA1.hexdigest(LOCK_SCRIPT)
+        @pttl_script_sha   = Digest::SHA1.hexdigest(PTTL_SCRIPT)
```

Comment:
could you not precalculate these at module load time instead of for each Redlock::Client initialization?

Reply:
Thanks for the quick replies! Yup this could happen at any point. To do that it seems like the script constants, e.g. `UNLOCK_SCRIPT`, should also be moved up to the `Redlock` module level.

It could be nice to extract that to a new file `scripts.rb` or something, and do this:

```
module Redlock
  module Scripts
    UNLOCK_SCRIPT = ...
    UNLOCK_SCRIPT_SHA = Digest::SHA1.hexdigest(UNLOCK_SCRIPT)
    ...
```

then in `client.rb`
```
module Redlock
  include Scripts

  class Client
    ...
```

Is that along the lines of what you were thinking?

Reply:
Sounds good, yes